app/src/main/java/unipd/se18/ocrcamera/TextExtractor.java

```diff
@@ -0,0 +1,240 @@
+                                + "-----      RECOGNIZED TEXT       -----"
+                                + "\n" + firebaseVisionText.getText() + "\n"
+                                + "----- END OF THE RECOGNIZED TEXT -----");
+                        // Extraction ended - Analogous to signal
+                        extraction.countDown(); // Luca Moroldo (g3)
+                    }
+                });
+        // Waits until the extraction ends
+        if (!firebaseVisionTextTask.isSuccessful()) {
+            try {
+                // Analogous to wait
+                extraction.await(); // Luca Moroldo (g3)
+            } catch (InterruptedException e) {
+                Log.e(TAG, methodTag + "Synchronizing problem -> "
+                        + "return null (nothing recognized)");
+                return null;
+            }
+
+        }
+
+        // Ends the time counter - useful for tests
+        long afterWaiting = java.lang.System.currentTimeMillis();
+
+        Log.i(TAG, methodTag + "text extracted in "
+                + (afterWaiting - beforeWaiting) + " ms");
+
+        // Returns the FirebaseVisionText recognized by the task
+        return firebaseVisionTextTask.getResult();
+    }
+
+    /**
+     * Produces a String from a FirebaseVisionText
+     * More details at: {@link FirebaseVisionText.TextBlock#getText()}.
+     * @param firebaseVisionTextExtracted The result of the FirebaseVisionText extraction
+     * @return String extracted by the FirebaseVisionText result
+     * @author Pietro Prandini (g2)
+     */
+    private String extractString(FirebaseVisionText firebaseVisionTextExtracted) {
+        // Orders the blocks (requested by the issue #18 but not so useful at the moment)
+        ArrayList<FirebaseVisionText.TextBlock> textBlocks =
+                sortBlocks(firebaseVisionTextExtracted);
+
+        // Prepares the String with the text of the blocks identified by the Firebase process
+        StringBuilder text = new StringBuilder();
+        for(FirebaseVisionText.TextBlock block: textBlocks) {
+            text.append(block.getText()).append("\n");
+        }
+        String extractedText = text.toString();
+
+        // Returns the String
+        return extractedText;
+    }
+
+    /*
+    Sorting methods are not so useful at the moment.
+    There is an issue (#18) that request this method for a possible future use.
+     */
+
+    /**
+     * Sorts the blocks recognized
+     * More details at: {@link FirebaseVisionText.TextBlock#getTextBlocks()}.
+     * @param OCRResult FirebaseVisionText object produced by an OCR recognition
+     * @return An ArrayList of FirebaseVisionText sorted
+     * @author Pietro Prandini (g2)
+     */
+    private ArrayList<FirebaseVisionText.TextBlock> sortBlocks(FirebaseVisionText OCRResult) {
+        ArrayList<FirebaseVisionText.TextBlock> OCRBlocks =
+                new ArrayList<>(OCRResult.getTextBlocks());
+        OCRBlocks = sortBlocksY(OCRBlocks);
```

Comment:
Why did you split these 2 methods? For me they are not complicated to be splitted.

Reply:
I've split this method because the method for sorting could be change or basically in the near future could be useful sorting the blocks from left to right and not only from top to bottom.
So you suggest anyway to merge the methods or to left the methods split?
Thanks in advance.

Reply:
Oh ok, no its fine. This is reasonable solution for the future

Reply:
Thanks for reviewing.